feat: introduce Horizontal Navigation pattern

InnaAtanasova opened the pull request:
## Related Issue
Closes SAP/fundamental-styles#2210


BREAKING CHANGE: 
- Added an additional modifier class `fd-icon-tab-bar__item--overflow` to `fd-icon-tab-bar__item` for the `li` element wrapping the overflow button
- Added an additional modifier class `fd-icon-tab-bar__popover-body` to `fd-popover__body` for the overflow

## Description
 Horizontal Navigation is a composite component built on extended functionality of Universal IconTabBar and Tool Header.

#### Please check whether the PR fulfills the following requirements

1. The output matches the design specs
- [x] All values are in `rem`
- [x] Text elements follow the truncation rules
- [x] hover state of the element follow design spec
- [x] focus state of the element follow design spec
- [x] active state of the element follow design spec
- [x] selected state of the element follow design spec
- [x] selected hover state of the element follow design spec
- [x] pressed state of the element follow design spec
- [x] Responsiveness rules - the component has modifier classes for all breakpoints
- [x] Includes Compact/Cosy/Tablet design
- [x] RTL support
2. The code follows fundamental-styles code standards and style
- [x] only one top level `fd-*` class is used in the file
- [x] BEM naming convention is used
- [x] Mixins are used for repeatable code (`fd-rtl`, `fd-ellipsis`, `fd-flex`, `fd-selected`, `fd-focus`, ect.)
- [x] A11y support - keyboard support, screenreader support, proper ARIA attributes, etc.
- [x] `fd-reset()` mixin is applied to all elements
- [x] Variables are used, if some value is used more than twice.
- [x] Checked if current components can be reused, instead of having new code.
3. Testing
- [x] tested Storybook examples with "CSS Resources" `normalize` option 
- [x] tested Storybook examples with "CSS Resources" `unnormalize` option 
- [x] Verified all styles in IE11
- [x] Updated tests
- [x] last commit message should have `[ci visual]` so it can trigger chromatic visual regression (e.g. `test: run chromatic visual regression [ci visual]`)
4. Documentation
- [x] Storybook documentation has been created/updated
- [x] Breaking Changes [wiki](https://github.com/SAP/fundamental-styles/wiki/Breaking-Changes) has been updated in case of breaking changes.

platon-rov commented on src/icon-tab-bar.scss:
```diff
@@ -132,10 +132,25 @@ $fd-icon-tab-bar-semantic-values: (
+  }
+}
+
+@mixin setNavigationTabColors($color: var(--sapShell_Navigation_Selected_TextColor)) {
```
I see that every other mixin named in kebab-case and only this one in camelCase, should it be unified?

InnaAtanasova replied:
you are absolutely right. I don't know where my mind was when I did this :D Probably on vacation. Thanks for noticing.Fixing it right away  :)

InnaAtanasova commented:
BREAKING CHANGE:
Added an additional modifier class fd-icon-tab-bar__item--overflow to fd-icon-tab-bar__item for the li element wrapping the overflow button
Added an additional modifier class fd-icon-tab-bar__popover-body to fd-popover__body for the overflow
